apis/v1alpha1/gateway_types.go

```diff
@@ -298,6 +298,12 @@ const (
 // * "TCP"
 // * "UDP"
 //
+// Implementations can define their own protocols if one is not already defined.
+// Un-prefixed names are reserved for IANA standard service names (as per
+// RFC-6335 and http://www.iana.org/assignments/service-names).
+// Non-standard protocols should use prefixed names such as
+// mycompany.com/my-custom-protocol.
```

Comment:
Do we need to clarify the support level here? Assuming this is custom?

Reply:
Updated.

Reply:
I think that this needs a broader discussion and concept. The protocols we currently specify are spread across multiple OSI layers (e.g. we have both "UDP" and "HTTP"), and can be protocol families (e.g. "HTTPS" probably means HTTP/1.1 and HTTP/2, possibly HTTP/3).

The IANA service-names registry is just UDP/TCP port registrations. There's no "h2" or "GRPC" for example.

Reply:
@jpeach Those are good points, do you know of any better lists we could use here?

Reply:
Consider splitting `ProtocolType` into `AppProtocolType` that contains application layer protocols (i.e. HTTP) and `TransportProtocolType` that contains transport layer protocols (i.e. TCP).

Reply:
At least short term, what if we remove the RFC reference entirely and simplify the options to:
- Protocols we've listed and officially support
- Custom protocols can be specified with domain prefixes

We can always expand our officially supported list in the future but this feels like a reasonable compromise for now.

Reply:
@robscott I like that approach. I've updated the PR to reflect that.